src/lib/support/PersistedCounter.cpp

```diff
@@ -51,7 +51,7 @@ PersistedCounter::Init(const chip::Platform::PersistedStorage::Key aId, uint32_t
     ChipLogDetail(EventLogging, "PersistedCounter::Init() aEpoch 0x%x startValue 0x%x", aEpoch, startValue);
 #endif
 
-    err = PersistNextEpochStart(startValue + aEpoch);
+    err = PersistNextEpochStart(startValue);
```

Comment:
was this a bug in existing code? I thought we had unit tests ... I am surprised it was not caught. Could you add a unit test that would catch the issue?

Reply:
The main problem in the PersistedCounter, it assumes that consumers will need to increment the counter on each initialization which is a not valid assumption in all scenarios, so i limited the incrementing to when we use the advance method only to avoid any implicit logic.

Reply:
I am not convinced this is a complete fix/change:

- Advance() still moves the stored value in an epoch
- without this logic, unexpected reboots will move back to 'Init' values, so we lose any internal 'Advance' calls completely and ids are reused (if we do ID for privacy, reuse sounds bad)

Reply:
As discussed offline, i will add a new class/counter for the "RotationCounter" to support incrementing only on demand